### ISSUE TITLE
help request: The control layer of Apisix cannot use the etcd cluster of ipv6

### ISSUE THREAD

**Issue by andychyiyi** [apache/apisix#7997]
### Description

- Error starting apisix

```
[root@master01 ~]# apisix start
/usr/local/openresty/luajit/bin/luajit /usr/local/apisix/apisix/cli/apisix.lua start
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379/version' error, Address family for hostname not supported, retry time=1
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379/version' error, Address family for hostname not supported, retry time=2
request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379/version' error, Address family for hostname not supported
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b22]:2379/version' error, Address family for hostname not supported, retry time=1
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b22]:2379/version' error, Address family for hostname not supported, retry time=2
request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b22]:2379/version' error, Address family for hostname not supported
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b23]:2379/version' error, Address family for hostname not supported, retry time=1
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b23]:2379/version' error, Address family for hostname not supported, retry time=2
request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b23]:2379/version' error, Address family for hostname not supported
all etcd nodes are unavailable
```

- The address of direct access to ipv6 is incorrect. It is speculated that Apisix uses a similar method to perform etcd health check internally. I hope to perform troubleshooting

```
[root@master01 ~]# curl http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379/version
curl: (3) [globbing] error: bad range specification after pos 9
```

- Using ipv6, the address of direct access to ipv6 is correct

```
[root@master01 ~]# curl -g6 http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379/version
{"etcdserver":"3.5.0","etcdcluster":"3.5.0"}[root@master01 ~]#
```

- The etcd address in the configuration file is the address of ipv6
```
[root@master01 ~]# cat /usr/local/apisix/conf/config.yaml
...Omission
etcd:
  host:
    - http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379
    - http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b22]:2379
    - http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b23]:2379
apisix:
  admin_key:
    - name: admin
      key: edd1c9f034335f136f87ad84b625c8f1  # using fixed API token has security risk, please update it when you deploy to production environment
      role: admin

```

My requirement is that ipv6 is used in all components, and the running environment does not have ipv4
I hope to draw your attention. Is there a good solution?

### Environment

- APISIX version (run `apisix version`):2.15.0
- Operating system (run `uname -a`):GNU/Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):1.21.4.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):3.5.0
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by tzssangglass** [comment]
ref: https://github.com/apache/apisix/issues/7100#issuecomment-1153706828

It seems necessary to make etcd's healthcheck support ipv6. cc @spacewander @soulbird

**Comment by tzssangglass** [comment]
Or you can wait and try the 3.0 version, the health check function of etcd will be disabled in 3.0
ref: https://github.com/apache/apisix/blob/3c18031c6e70daa15db72fe06ae8644b04b4481e/apisix/core/etcd.lua#L139-L141

**Comment by spacewander** [comment]
We still need to fix it in 3.0 as the health check logic is still in use on the CP side.

**Comment by andychyiyi** [comment]
Hello, I have simply tested the version of apisix-v2.99.0. This problem still exists. Would you like to draw attention to the fact that the subsequent official version of v3.0 will solve this problem? @spacewander @tzssangglass

**Comment by tzssangglass** [comment]
> Will the subsequent official version v3.0 solve this problem

Fix it after the 3.0 version is released? What are your opinions? cc @spacewander @soulbird

**Comment by soulbird** [comment]
3.0 being a big release, I think we'd better fix it before release so that we can make APISIX all run under an IPV6 environment, which is very helpful for our publicity.

**Comment by andychyiyi** [comment]
I hope the configuration mode of ipv4/6 here is still the same
Look forward to a better apisix, and wait for the update.

**Comment by andychyiyi** [comment]
@soulbird @tzssangglass 
```
[root@master01 ~]# apisix version
/usr/local/openresty/luajit/bin/luajit /usr/local/apisix/apisix/cli/apisix.lua version
3.0.0
[root@master01 ~]# apisix start
/usr/local/openresty/luajit/bin/luajit /usr/local/apisix/apisix/cli/apisix.lua start
Warning! Current maximum number of open file descriptors [1024] is not greater than 1024, please increase user limits by execute 'ulimit -n <new user limits>' , otherwise the performance is low.
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379/version' error, Address family for hostname not supported, retry time=1
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379/version' error, Address family for hostname not supported, retry time=2
request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b21]:2379/version' error, Address family for hostname not supported
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b22]:2379/version' error, Address family for hostname not supported, retry time=1
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b22]:2379/version' error, Address family for hostname not supported, retry time=2
request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b22]:2379/version' error, Address family for hostname not supported
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b23]:2379/version' error, Address family for hostname not supported, retry time=1
Warning! Request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b23]:2379/version' error, Address family for hostname not supported, retry time=2
request etcd endpoint 'http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b23]:2379/version' error, Address family for hostname not supported
all etcd nodes are unavailable
[root@master01 ~]# curl http://[fd15:4ba5:5a2b:1008:20c:29ff:fe7d:4b23]:2379/version -g6
{"etcdserver":"3.5.4","etcdcluster":"3.5.0"}[root@master01 ~]#
[root@master01 ~]#
[root@master01 ~]#
```

Hello, the etcd health check of apisixv3.0 has not been repaired by IPV6
Did you forget to fix it, or was it because my configuration was not enabled?
After installation, I only modified the default etcd address.
I hope you can give me some advice...

**Comment by tzssangglass** [comment]
> Hello, the etcd health check of apisixv3.0 has not been repaired by IPV6

Just fixed, not in the 3.0.0 release. ref: #8245

**Comment by andychyiyi** [comment]
I see. Thank you. I'll try again
